public/tabs/history/chat_history_page.tsx

```diff
@@ -96,17 +101,44 @@ export const ChatHistoryPage: React.FC<ChatHistoryPageProps> = React.memo((props
     [searchName]
   );
 
-  useEffect(() => {
-    if (props.shouldRefresh) services.conversations.reload();
+  useUpdateEffect(() => {
+    if (!props.shouldRefresh) {
+      return;
+    }
+    services.conversations.reload();
+    return () => {
+      services.conversations.abortController?.abort();
+    };
   }, [props.shouldRefresh, services.conversations]);
 
   useEffect(() => {
     services.conversations.load(bulkGetOptions);
+    const subscription = services.dataSource.dataSourceIdUpdates$.subscribe(() => {
+      // Load conversations directly if options are default values
+      if (!bulkGetOptions.search && bulkGetOptions.page === 1) {
+        services.conversations.reload();
+        return;
+      }
+      // Reset changes to default, these operations will trigger load after state updates
+      if (bulkGetOptions.search) {
+        setSearchName('');
+        setDebouncedSearchName('');
+      }
+      if (bulkGetOptions.page !== 1) {
+        setPageIndex(0);
+      }
+    });
     return () => {
       services.conversations.abortController?.abort();
+      subscription.unsubscribe();
     };
   }, [services.conversations, bulkGetOptions]);
```

Comment:
```suggestion
  const dataSourceUpdate = useObservable(services.dataSource.dataSourceIdUpdates$);

  useEffect(() => {
    setSearchName('');
    setDebouncedSearchName('');
    setPageIndex(0);
  }, [dataSourceUpdate]);

  useEffect(() => {
    services.conversations.load(bulkGetOptions);
    return () => {
      services.conversations.abortController?.abort();
    };
  }, [services.conversations, bulkGetOptions]);
```
What about changing to like this?